### ISSUE TITLE
Large responses are not completely sent with 'Connection: close'

### ISSUE THREAD

**Issue by minus7** [ktorio/ktor#534]
When a ktor server sends a large amount of data over the network and the client specified `Connection: close`, the transfer will not complete.

```
curl http://poseidon:2000/ -H 'Connection: close' > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  8 1000k    8 85432    0     0  3776k      0 --:--:-- --:--:-- --:--:-- 3792k
curl: (18) transfer closed with 938568 bytes remaining to read
```

This doesn't happen on localhost connections, only on those going over at least a LAN (maybe VMs too).

The following code in combination with the curl command reproduces the problem:
```kotlin
import io.ktor.application.call
import io.ktor.response.respond
import io.ktor.routing.get
import io.ktor.routing.routing
import io.ktor.server.engine.embeddedServer
import io.ktor.server.netty.Netty
import java.io.File

fun main(args: Array<String>) {
  val server = embeddedServer(Netty, 2000) {
    routing {
      get("/") {
        val data = ByteArray(1024 * 1024)
        File("/dev/urandom").inputStream().read(data)
        call.respond(data)
      }
    }
  }
  server.start()
}
```

**Comment by soywiz** [comment]
In the case it helps somehow:
I had a similar problem too (reproduced in 0.9.3 in a docker instance serving to the Internet), using Netty I was sending a big file from a ByteArray (~10MB) with `respondBytes` and the connection was consistently being closed earlier than expected (at 1.2MB or something like that) both wget and chrome failed. I circumvented the problem by storing the bytes inside a file and serving the file with `respondFile`. Not tested with 0.9.4 dev builds.

**Comment by jonstacks** [comment]
We just hit this as well in production when trying to transfer a payload of about 1M. We are using nginx with the `proxy_pass` directive and by default it sets the header `Connection:  close`. We will mostly likely be adding this in the nginx config as a workaround:

```
proxy_set_header Connection "keep-alive";
```

We ended up trying things like turning off `proxy_buffering` and that seemed to work as well. Hope this helps someone else.

**Comment by cy6erGn0m** [comment]
Fixed in 0.9.4

**Comment by minus7** [comment]
Thanks!
